src/live_restore/live_restore_fs.c

```diff
@@ -589,7 +592,24 @@ __live_restore_fs_fill_holes_on_file_close(WT_FILE_HANDLE *fh, WT_SESSION *wt_se
         __wt_verbose_debug3((WT_SESSION_IMPL *)wt_session, WT_VERB_FILEOPS,
           "Found hole in %s at %ld-%ld during file close. Filling", fh->name, hole->off,
           WT_EXTENT_END(hole));
-        WT_RET(__live_restore_fh_read(fh, wt_session, hole->off, hole->len, buf));
+
+        off_start = hole->off;
+        off_end = WT_EXTENT_END(hole);
+
+        /*
+         * When encountering a large hole, break the read into small chunks. Split the hole into
+         * n chunks: the first n - 1 chunks will read a full WT_LIVE_RESTORE_READ_SIZE buffer, and
+         * the last chunk reads the remaining data.
+         */
+        while (off_start + WT_LIVE_RESTORE_READ_SIZE <= off_end) {
```

Comment:
I would change this logic to be based off read_len:
```
read_len = hole->len;
while (read_len > 0) {
     WT_RET(__live_restore_fh_read(fh, wt_session, off_start, read_len, buf));
     off_start += read_len;
     read_len = off_end - off_start > WT_LIVE_RESTORE_READ_SIZE ? WT_LIVE_RESTORE_READ_SIZE : off_end - off_start;
}

/* Detect any issues with the read_len logic. */
WT_ASSERT(session, read_len == 0);
```
If `read_len` is signed we'll need to make it signed.

Reply:
Thanks Luke for the suggestion. How about the below one:
```
read_len = hole->len;
while (read_len > WT_LIVE_RESTORE_READ_SIZE) {
     WT_RET(__live_restore_fh_read(fh, wt_session, off_start, WT_LIVE_RESTORE_READ_SIZE, buf));
     off_start += WT_LIVE_RESTORE_READ_SIZE;
     read_len -= WT_LIVE_RESTORE_READ_SIZE;
}
WT_RET(__live_restore_fh_read(fh, wt_session, off_start, read_len, buf));
```
So we don't need `off_end` anymore. 
The other thing is that we can eliminate the ternary operator. I'm not sure if wiredtiger is being nitpicky to the point of preferring simple assignment over the ternary operator purely because it's slightly faster. However, if that's the case, using simple assignment might be reasonable, though the performance improvement is likely negligible.  

Correct me if I'm wrong.

Reply:
I have a second implementation suggestion, which depends on the fact that we're actually modifying the linked list as we read in holes: 
```
/*
 * __live_restore_fs_fill_holes_on_file_close --
 *     On file close make sure we've copied across all data from source to destination. This means
 *     there are no holes in the destination file's extent list. If we find one promote read the
 *     content into the destination.
 *
 * NOTE!! This assumes there cannot be holes in source, and that any truncates/extensions of the
 *     destination file are already handled elsewhere.
 */
static int
__live_restore_fs_fill_holes_on_file_close(WT_FILE_HANDLE *fh, WT_SESSION *wt_session)
{
/*
 * Holes can be large, potentially the size of an entire file. When we find a large hole we'll read
 * it in 4KB chunks.
 */
#define WT_LIVE_RESTORE_READ_SIZE ((size_t)(4 * WT_KILOBYTE))
    char buf[WT_LIVE_RESTORE_READ_SIZE];
    WT_LIVE_RESTORE_FILE_HANDLE *lr_fh = (WT_LIVE_RESTORE_FILE_HANDLE *)fh
    WT_LIVE_RESTORE_HOLE_LIST *hole, *last_hole = NULL;

    while (lr_fh->destination.hole_list != NULL) {
        hole = lr_fh->destination.hole_list;
        if (last_hole != hole) {
            __wt_verbose_debug3((WT_SESSION_IMPL *)wt_session, WT_VERB_FILEOPS,
              "Found hole in %s at %" PRId64 "-%" PRId64 " during file close. Filling", fh->name,
              hole->off, WT_EXTENT_END(hole));
        }
        /*
         * When encountering a large hole, break the read into small chunks. Split the hole into
         * n chunks: the first n - 1 chunks will read a full WT_LIVE_RESTORE_READ_SIZE buffer, and
         * the last chunk reads the remaining data. This loop is a not obvious, effectively the read
         * is shrinking the hole in the stack below us. This is why we always read from the start at
         * the beginning of the loop.
         */
        WT_RET(__live_restore_fh_read(fh, wt_session, off_start, hole->len > WT_LIVE_RESTORE_READ_SIZE ? WT_LIVE_RESTORE_READ_SIZE : hole->len, buf));
        last_hole = hole;
    }
    return (0);
}
```

This eliminates the needs for a second loop.